defold: init at 1.9.8

musjj opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Closes https://github.com/NixOS/nixpkgs/issues/95761

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

musjj committed:
defold: init at 1.9.8

Sigmanificient commented on pkgs/by-name/de/defold/package.nix:
```diff
@@ -0,0 +1,107 @@
+  version = "1.9.8";
+
+  src =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
```
Why this condition if you set `platforms = [ "x86_64-linux" ];`?

Sigmanificient commented on pkgs/by-name/de/defold/package.nix:
```diff
@@ -0,0 +1,107 @@
+    jdk
+    libgcc.lib
+    libGL
+    xorg.libX11
+    xorg.libXext
+    xorg.libXi
+    xorg.libXrender
+    xorg.libXtst
+    xorg.libXxf86vm
```
You can strip all `xorg.` prefixes

Sigmanificient commented on pkgs/by-name/de/defold/package.nix:
```diff
@@ -0,0 +1,107 @@
+  '';
+
+  meta = with lib; {
+    description = "Defold is a completely free to use game engine for development of desktop, mobile and web games";
```
```suggestion
    description = "Game engine for development of desktop, mobile and web games";
```

Sigmanificient commented on pkgs/by-name/de/defold/package.nix:
```diff
@@ -0,0 +1,107 @@
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
```suggestion
  meta = {
```
- #208242

musjj committed:
defold: refactor to a FHS env

Defold downloads additional binaries on run-time, so it's impossible to
patch everything on build-time.

musjj committed:
defold: remove unnecessary platform checks

musjj committed:
defold: refactor xorg prefixes

musjj committed:
defold: less redundant description

musjj committed:
defold: smaller "with lib" scope

musjj committed:
defold: remove unnecessary copying

musjj committed:
defold: fix name overlap with license

musjj commented:
> Error: fatal: Could not read from remote repository.

Not sure what's going on, but I'm getting the same error locally. Is there a problem with GitHub?

musjj committed:
defold: remove Game from desktop category

musjj committed:
defold: remove unnecessary pre/postinstall hooks

alyssais commented on lib/licenses.nix:
```diff
@@ -445,6 +445,13 @@ lib.mapAttrs mkLicense ({
   };
 
+  defold = {
+    fullName = "Defold License";
```
Given this will only ever be used by one package, maybe it'd make more sense to inline it?

musjj replied:
Alright, I've inlined it.

musjj committed:
defold: inline defold license

qweered commented:
It would be nice if you add support for other platforms and update script

qweered commented:
Also it is better to build from source than unpack

qweered reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.
